### DISCUSSION TITLE
NextJS Deployment error because of node version 18

### DISCUSSION THREAD

**Discussion by jessecheu** [vercel/vercel#9131] [Help]
Hello,

Currently, Vercel can't deploy NextJS apps with node version "18.x+" and must use "16.x". 
I get the error:
`<html>
<body>

00:00:42.508 | Retrieving list of deployment files...
-- | --
00:00:42.960 | Previous build caches not available
00:00:43.013 | Downloading 52 deployment files...
00:00:43.590 | Running "vercel build"
00:00:44.043 | Vercel CLI 28.10.0
00:00:44.311 | Warning: Due to "engines": { "node": "18.x" } in your `package.json` file, the Node.js Version defined in your Project Settings ("16.x") will not apply. Learn More: http://vercel.link/node-version
00:00:44.319 | Installing dependencies...
00:00:47.838 | npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
00:01:08.496 | npm ERR! code 1
00:01:08.496 | npm ERR! path /vercel/path0/node_modules/esbuild
00:01:08.496 | npm ERR! command failed
00:01:08.496 | npm ERR! command sh -c -- node install.js
00:01:08.496 | npm ERR! node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
00:01:08.497 | npm ERR! node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
00:01:08.497 |  
00:01:08.497 | npm ERR! A complete log of this run can be found in:
00:01:08.497 | npm ERR!     /vercel/.npm/_logs/2022-12-26T08_00_44_537Z-debug-0.log
00:01:08.518 | Error: Command "npm install" exited with 1

</body>
</html>`
One fix is changing package.json to
 
`"node": "^16.16.0" `
in the dependencies. 

and adding 

```
"engines": {
    "node": "16.x"
  }
```

**Comment by amyegan** [comment]
Hi @jessecdob! Sorry for the delay. It looks like you have already implemented an appropriate workaround, but I'm not sure what was causing the version disagreement. 

Do you have a public repository or [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share? If you can share some more information about your deployment workflow or any build tools you're using, that would also be helpful. 🙏
